src/plugins/discover/public/application/main/hooks/use_text_based_query_language.ts

```diff
@@ -80,24 +81,14 @@ export function useTextBasedQueryLanguage({
           }
         }
         const indexPatternFromQuery = getIndexPatternFromSQLQuery(query.sql);
-        const internalState = stateContainer.internalState.getState();
-        const dataViewList = [...internalState.savedDataViews, ...internalState.adHocDataViews];
-        let dataViewObj = dataViewList.find(({ title }) => title === indexPatternFromQuery);
 
-        // no dataview found but the index pattern is valid
-        // create an adhoc instance instead
-        if (!dataViewObj) {
-          dataViewObj = await dataViews.create({
-            title: indexPatternFromQuery,
-          });
-          stateContainer.internalState.transitions.setAdHocDataViews([dataViewObj]);
+        const dataViewObj = await dataViews.create({
+          title: indexPatternFromQuery,
+        });
+        stateContainer.internalState.transitions.setAdHocDataViews([dataViewObj]);
 
-          if (dataViewObj.fields.getByName('@timestamp')?.type === 'date') {
-            dataViewObj.timeFieldName = '@timestamp';
-          } else if (dataViewObj.fields.getByType('date')?.length) {
-            const dateFields = dataViewObj.fields.getByType('date');
-            dataViewObj.timeFieldName = dateFields[0].name;
-          }
```

Comment:
QQ, looking at the code, you no longer want to take care of dataViews that have date fields but no '@timestamp' named one?

Reply:
Exactly, it is @timestamp or full dataset

Reply:
thx!